base/abstractarraymath.jl

```diff
@@ -91,6 +91,10 @@ imag{T<:Real}(x::AbstractArray{T}) = zero(x)
 
 # index A[:,:,...,i,:,:,...] where "i" is in dimension "d"
 
+_slice_inds(new_inds::Tuple, inds::Tuple, d, i) =
+    _slice_inds((new_inds..., length(new_inds) + 1 == d ? i : first(inds)), tail(inds), d, i)
+_slice_inds(new_inds::Tuple, inds::Tuple{}, d, i) = new_inds
```

Comment:
Seems like a good idea. You'd want the `v` to come before `n` in the argument order, though, for consistency with `setindex!`.